src/ape/contracts/base.py

```diff
@@ -842,6 +843,31 @@ def at(self, address: AddressType, txn_hash: Optional[str] = None) -> ContractIn
             address, self.contract_type, txn_hash=txn_hash
         )
 
+    @cached_property
+    def source_path(self) -> Optional[Path]:
+        """
+        Returns the path to the local contract if determined that this container
+        belongs to the active project.
+        """
+        if self.contract_type.name is not None:
+            contract_name = str(self.contract_type.name)
+
+        contract_container = self.project_manager._get_contract(contract_name)
+
+        if (
+            contract_container
+            and contract_container.contract_type.runtime_bytecode
```

Comment:
We should pull these out as local variables to clean this up and is more type adherent:

```
if not contract_container:
    return None
    
contract_bytecode = contract_container.contract_type.runtime_bytecode
self_bytecode = self.contract_type.runtime_bytecode.bytecode

if (
    contract_bytecode
    and self_bytecode
    and self.contract_type.source_id
    and contract_bytecode == self_bytecode
):
```